### DISCUSSION TITLE
Are PRs for Linux version accepted?

### DISCUSSION THREAD

**Discussion by badcel** [DevToys-app/DevToys#1157] [Q&A] [answered]
Do you accept PRs updating GirCore and the related Linux C# code? I think there are some things that could be updated as a new version ([0.5.0](https://github.com/gircore/gir.core/releases/tag/0.5.0)) just got released.

**Comment by veler** [comment, marked as answer]
Yes! Please make sure to test it on a Debian-based OS. :)

**Reply by veler** [reply]
Oh I just saw you're the author of GirCore. Good job and thank you for making this library! We struggled at finding a GTK wrapper for .NET that was fitting our needs 😅

**Reply by badcel** [reply]
Thank you. GirCore is still in its early stages and I'm thankful that you use it as a way to create DevToys for Linux 😎. It's a pretty good show case for what is possible.

I assume Debian based means current Ubuntu 24.04?

**Reply by veler** [reply]
My pleasure 😁
Yes, by Debian-based I meant any distro based on it like Ubuntu, Debian itself, Raspberry Pi OS...etc. No need to test them all, just one should be enough.

Have fun! I do admit I'm a noob at Linux, and making DevToys working on it was quite challenging, and it's a little unstable compared to Windows and Mac. Appreciate your effort 😁
Feel free to suggest any improvement / refactoring / bug fix on Linux to make it better.

**Reply by badcel** [reply]
@veler: Do you have some kind of reference linux distribution where everything should just work, meaning a little more specific than "Debian based"? (What did you use yourself for the linux development?)

I tried running DevToys in a Toolbox-Container (no virtual machine) which was based on Ubuntu 23.10. and the Icons were missing. So I blamed the Toolbox installation and set up a virtual Ubuntu 24.04. machine and ran the software and the icons were missing, too.

On the other hand running it under Fedora 40 displays the icons like expected. Unfortunately I don't have time to work on such issues as I would need to research the topic, too. I just wanted to create some PRs to improve the GirCore side of things to spare you some GirCore research time.

Alternatively are you okay with PRs if the icons in my test environment were missing before and after the changes?

Edit: Selecting the "Settings" under Ubuntu 24.04 crashes for me while it works with fedora. So I would think 24.04 is no suitable test environment for me.

**Reply by veler** [reply]
That's interesting!thanks for sharing. On my side I developed the Linux version entirely on Ubuntu and Debian. I know someone made it running on Arch and Windows subsystem for Linux but didn't try myself.

I'm fine with you making a PR of the icon doesn't appear. I will test it on my side and see how it is.

Also, for your information, the icon in the task bar is defined through the ".desktop" file that is installed by the Deb file we generate (I got a private CI for it). When Debugging locally, I never got the icon to work.

**Reply by badcel** [reply]
Now I created a debian 12 container and it works like expected. This is fine for me. So I can check my changes on fedora during development and additionally in a debian 12 environment afterwards. Nevertheless it is nice to know that you check the code changes.
